src/ota_providers.sh

```
@@ -0,0 +1,118 @@
+    return 1
+  }
+  latest_version="${release_metadata%%[[:space:]]*}"
+  if [[ ! "${latest_version}" =~ ^[0-9]{8,14}$ ]]; then
+    echo "Error: GrapheneOS returned an invalid release version." >&2
+    return 1
+  fi
+
+  VERSION[GRAPHENEOS]="${GRAPHENEOS_VERSION:-${latest_version}}"
+  ROM_OTA_SHA256=""
+  GRAPHENEOS[OTA_TARGET]="${DEVICE_NAME}-${GRAPHENEOS[UPDATE_TYPE]}-${latest_version}"
+  GRAPHENEOS[OTA_URL]="${ROM_PROFILE[OTA_BASE_URL]}/${GRAPHENEOS[OTA_TARGET]}.zip"
```

Shellcheck (0.11.0)

[info] Possible misspelling: OTA_URL may not be assigned. Did you mean ota_url?

(SC2153)